### ISSUE TITLE
Test Failure under musl c in 2.0.0

### ISSUE THREAD

**Issue by itoffshore** [python-lz4/python-lz4#134]
Version `2.0.0` builds successfully under Alpine Linux (musl c) - but the tests fail at the end:

```
...............................................................................................................................................                     [ 95%]
tests/frame/test_frame_5.py F

================================================================================ FAILURES =================================================================================
_________________________________________________________________ test_frame_decompress_mem_usage[data0] __________________________________________________________________

data = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'

    def test_frame_decompress_mem_usage(data):
        tracemalloc = pytest.importorskip('tracemalloc')
    
        tracemalloc.start()
    
        compressed = lz4.frame.compress(data)
        prev_snapshot = None
    
        for i in range(1000):
            decompressed = lz4.frame.decompress(compressed)  # noqa: F841
    
            if i % 100 == 0:
                snapshot = tracemalloc.take_snapshot()
    
                if prev_snapshot:
                    stats = snapshot.compare_to(prev_snapshot, 'lineno')
>                   assert stats[0].size_diff < (1024 * 4)
E                   AssertionError: assert 11264 < (1024 * 4)
E                    +  where 11264 = <StatisticDiff traceback=<Traceback (<Frame filename='/usr/lib/python3.6/tracemalloc.py' lineno=113>,)> size=11264 (+11264) count=128 (+128)>.size_diff

compressed = b'\x04"M\x18H@\x00\x00\x10\x00\x00\x00\x00\x00g\x0b\x01\x00\x00\x1fa\x01\x00\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\x...\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xff\xe8Paaaaa\x00\x00\x00\x00'
data       = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
decompressed = b'aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa...aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa'
i          = 200
prev_snapshot = <tracemalloc.Snapshot object at 0x7f661e2d40f0>
snapshot   = <tracemalloc.Snapshot object at 0x7f661e2d43c8>
stats      = [<StatisticDiff traceback=<Traceback (<Frame filename='/usr/lib/python3.6/tracemalloc.py' lineno=113>,)> size=11264 (+...src/lz4-2.0.0/.eggs/pytest-3.6.1-py3.6.egg/_pytest/python.py' lineno=263>,)> size=842040 (-648) count=11695 (-9)>, ...]
tracemalloc = <module 'tracemalloc' from '/usr/lib/python3.6/tracemalloc.py'>

tests/frame/test_frame_5.py:35: AssertionError
=========================================================== 1 failed, 18108 passed, 1 skipped in 299.80 seconds ===========================================================
>>> ERROR: py-lz4*: check failed
>>> ERROR: py-lz4: all failed
```

**Comment by jonathanunderwood** [comment]
Well, that's strange - the frame code hasn't been touched in this release, other than a bugfix for Python 2,7 only, which you're not using. Are you sure version 1.1.0 is still passing OK? If so, this is probably an issue with lz4 1.8.2.

**Comment by jonathanunderwood** [comment]
If you revert commit 791d72f1057da54031547313d, does the test pass?

**Comment by jonathanunderwood** [comment]
Also: is there any CI platform for Alpine that could be integrated such that we catch things earlier - I am already building each commit on Travis and Appveyor, but obviously neither of these are building against musl.

**Comment by itoffshore** [comment]
we run the included unit tests for all package updates now under Travis

I'm unable to apply https://github.com/python-lz4/python-lz4/commit/791d72f1057da54031547313d0d09bb93789264c.diff

`Reversed (or previously applied) patch detected!  Assume -R? [n] `

**Comment by jonathanunderwood** [comment]
> I'm unable to apply https://github.com/python-lz4/python-lz4/commit/791d72f1057da54031547313d0d09bb93789264c.diff

Yes - I suggested reverting it, rather than applying it :)

You could either git clone the repo, and do a git revert. Or, if you want to use patch, use the -R flag (or answer yes to the message you see).

**Comment by jonathanunderwood** [comment]
> we run the included unit tests for all package updates now under Travis

Where can I see the relevant .travis.yml file for that?

**Comment by jonathanunderwood** [comment]
Also seeing the same test failure under glibc on Fedora 29: https://koji.fedoraproject.org/koji/watchlogs?taskID=27744033

This is with python 3.7.

**Comment by jonathanunderwood** [comment]
I have released version 2.0.1 which I believe should fix this problem.

**Comment by jonathanunderwood** [comment]
Closing this since I believe it's fixed in 2.0.1. Please re-open if not.

**Comment by itoffshore** [comment]
sorry for the late reply - [yes `2.0.1` fixes the tests](https://github.com/alpinelinux/aports/pull/4584)
